packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -0,0 +1,144 @@
+    logger: Logger,
+    discovery: PluginEndpointDiscovery,
+  ) {
+    this.config = config;
+    this.logger = logger;
+    this.discovery = discovery;
+  }
+
+  publish({ entity, directory }: PublishRequest): LocalPublishReturn {
+    const entityNamespace = entity.metadata.namespace ?? 'default';
+
+    const publishDir = resolvePackagePath(
+      '@backstage/plugin-techdocs-backend',
+      'static/docs',
+      entityNamespace,
+      entity.kind,
+      entity.metadata.name,
+    );
+
+    if (!fs.existsSync(publishDir)) {
+      this.logger.info(`Could not find ${publishDir}, creating the directory.`);
+      fs.mkdirSync(publishDir, { recursive: true });
+    }
+
+    return new Promise((resolve, reject) => {
+      fs.copy(directory, publishDir, err => {
+        if (err) {
+          this.logger.debug(
+            `Failed to copy docs from ${directory} to ${publishDir}`,
+          );
+          reject(err);
+        }
+
+        this.discovery
+          .getBaseUrl('techdocs')
+          .then(techdocsApiUrl => {
+            resolve({
+              remoteUrl: `${techdocsApiUrl}/static/docs/${entity.metadata.name}`,
+            });
+          })
+          .catch(reason => {
+            reject(reason);
+          });
+      });
+    });
+  }
+
+  fetchTechDocsMetadata(entityName: EntityName): Promise<string> {
+    return new Promise((resolve, reject) => {
+      this.discovery.getBaseUrl('techdocs').then(techdocsApiUrl => {
+        const storageUrl = new URL(
+          new URL(this.config.getString('techdocs.storageUrl')).pathname,
+          techdocsApiUrl,
+        ).toString();
+
+        const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
+        const metadataURL = `${storageUrl}/${entityRootDir}/techdocs_metadata.json`;
+        fetch(metadataURL)
+          .then(response =>
+            response
+              .json()
+              .then(techdocsMetadataJson => resolve(techdocsMetadataJson))
+              .catch(err => {
+                reject(
+                  `Unable to parse metadata JSON for ${entityRootDir}. Error: ${err}`,
+                );
+              }),
+          )
+          .catch(err => {
+            reject(
+              `Unable to fetch metadata for ${entityRootDir}. Error ${err}`,
+            );
+          });
+      });
+    });
+  }
+
+  docsRouter(): express.Handler {
+    return express.static(staticDocsDir);
+  }
+
+  async hasDocsBeenGenerated(entity: Entity): Promise<boolean> {
+    return new Promise(resolve => {
+      this.discovery.getBaseUrl('techdocs').then(techdocsApiUrl => {
+        const storageUrl = new URL(
+          new URL(this.config.getString('techdocs.storageUrl')).pathname,
+          techdocsApiUrl,
+        ).toString();
+
+        const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+        const indexHtmlUrl = `${storageUrl}/${entityRootDir}/index.html`;
+        fetch(indexHtmlUrl)
```

Comment:
Similar comment to GCS: perhaps simpler and more efficient to use `fs` or similar to check if the file exists instead of reading it?